sdk-utils/src/main/kotlin/com/rakuten/tech/mobile/sdkutils/StaticInfoUtil.kt

```diff
@@ -0,0 +1,191 @@
+    }
+
+    /**
+     * Get the SDK info.
+     *
+     * @param context the application context
+     * @return the sdk info
+     */
+    fun getSdkInfo(context: Context): String {
+        val sdkVersion = getModuleVersion(context, SdkInfoKey.MODULE_SDK)
+        return if (sdkVersion != null) {
+            SdkInfoKey.MODULE_SDK + "/" + sdkVersion
+        } else {
+            val sdkInfo = StringBuilder()
+            for (key in SdkInfoKey.KEYS) {
+                appendSdkModuleVersion(context, sdkInfo, key)
+            }
+            sdkInfo.toString()
+        }
+    }
+
+    private fun appendSdkModuleVersion(context: Context, sdkInfo: StringBuilder, moduleId: String) {
+        val version = getModuleVersion(context, moduleId)
+        if (version != null) {
+            if (sdkInfo.isNotEmpty()) {
+                sdkInfo.append("; ")
+            }
+            sdkInfo.append(moduleId).append('/').append(version)
+        }
+    }
+
+    /**
+     * Get the SDK info map.
+     *
+     * @param context the application context
+     * @return map of the sdk info
+     */
+    fun getSdkInfoMap(context: Context): HashMap<String, Any?> {
+        val infoMap = HashMap<String, Any?>()
+        for (key in SdkInfoKey.KEYS) {
+            val version = getModuleVersion(context, key)
+            if (version != null) {
+                infoMap[key] = getModuleVersion(context, key)
+            }
+        }
+        return infoMap
+    }
+
+    @SuppressWarnings("SwallowedException", "TooGenericExceptionCaught")
+    private fun getModuleVersion(context: Context, name: String): String? {
+        return try {
+            context.getString(
+                context.resources
+                    .getIdentifier(name + "__version", "string", context.packageName)
+            )
+        } catch (ex: Exception) {
+            null
+        }
+    }
+
+    // sdk info keys
+    internal object SdkInfoKey {
+        private const val MODULE_ANALYTICS = "analytics"
+        private const val MODULE_ANALYTICS_CORE = "analytics_core"
+        private const val MODULE_ANALYTICS_RAT = "analytics_rat"
+        private const val MODULE_ANALYTICS_IDTOKEN = "analytics_idtoken"
+        private const val MODULE_DISCOVER = "discover"
+        private const val MODULE_FEEDBACK = "feedback"
+        private const val MODULE_PING = "ping"
+        private const val MODULE_PUSH = "push"
+        private const val MODULE_IAM = "inappmessaging"
+        const val MODULE_SDK = "sdk"
+        val KEYS = arrayOf(/* Keys of all modules, MODULE_SDK ignored on purpose */
+            MODULE_ANALYTICS,
+            MODULE_ANALYTICS_CORE,
+            MODULE_ANALYTICS_RAT,
+            MODULE_ANALYTICS_IDTOKEN,
+            MODULE_DISCOVER,
+            MODULE_FEEDBACK,
+            MODULE_PING,
+            MODULE_PUSH,
+            MODULE_IAM
+        )
+    }
+
+    // app info keys
+    internal object AppInfoKey {
```

Comment:
Maybe better if we separate, app info, build info, and device info.

Reply:
updated